thrift/src/main/java/com/linecorp/armeria/server/thrift/THttpServiceBuilder.java

```diff
@@ -0,0 +1,171 @@
+ */
+public final class THttpServiceBuilder {
+
+    private static final SerializationFormat[] EMPTY_FORMATS = new SerializationFormat[0];
+    private final ImmutableListMultimap.Builder<String, Object> implementationsBuilder =
+            ImmutableListMultimap.builder();
+    private SerializationFormat defaultSerializationFormat = ThriftSerializationFormats.BINARY;
+    private Set<SerializationFormat> otherSerializationFormats = ThriftSerializationFormats.values();
+    @Nullable
+    private Function<? super RpcService, ? extends RpcService> decoratorFunction;
+
+    THttpServiceBuilder() { }
+
+    private static SerializationFormat[] newAllowedSerializationFormats(
+            SerializationFormat defaultSerializationFormat,
+            Iterable<SerializationFormat> otherAllowedSerializationFormats) {
+
+        requireNonNull(defaultSerializationFormat, "defaultSerializationFormat");
+        requireNonNull(otherAllowedSerializationFormats, "otherAllowedSerializationFormats");
+
+        final Set<SerializationFormat> set = new LinkedHashSet<>();
+        set.add(defaultSerializationFormat);
+        Iterables.addAll(set, otherAllowedSerializationFormats);
+        return set.toArray(EMPTY_FORMATS);
+    }
+
+    /**
+     * Adds a new {@code TMultiplexed} service to the builder.
+     *
+     * @param name name of the service.
+     * @param implementation an implementation of {@code *.Iface} or {@code *.AsyncIface} service interface
+     *                       generated by the Apache Thrift compiler.
+     */
+    public THttpServiceBuilder addService(String name, Object implementation) {
+        implementationsBuilder.put(name, implementation);
+        return this;
+    }
+
+    /**
+     * Add a new service implementation to the builder.
+     *
+     * @param implementation an implementation of {@code *.Iface} or {@code *.AsyncIface} service interface
+     *                       generated by the Apache Thrift compiler
+     */
+    public THttpServiceBuilder addService(Object implementation) {
+        implementationsBuilder.put("", implementation);
+        return this;
+    }
+
+    /**
+     * Adds other {@link SerializationFormat} to the builder. Current supported {@link SerializationFormat}s are
+     * {@link ThriftSerializationFormats#values()}. If nothing is specified then all the
+     * {@link SerializationFormat#values()}s are added.
+     *
+     * <p>Currently, the only way to specify a serialization format is by using the HTTP session
+     * protocol and setting the Content-Type header to the appropriate {@link SerializationFormat#mediaType()}.
+     */
+    public THttpServiceBuilder otherSerializationFormats(SerializationFormat... otherSerializationFormats) {
+        requireNonNull(otherSerializationFormats, "otherSerializationFormats");
+        this.otherSerializationFormats = new LinkedHashSet<>();
+        this.otherSerializationFormats.addAll(Arrays.asList(otherSerializationFormats));
```

Comment:
How about adding `otherSerializationFormats(Iterable<SerializationFormat>)` and delegating to it?

```java
requireNonNull(otherSerializationFormats, "otherSerializationFormats")
return otherSerializationFormats(ImmutableList.copyOf(otherSerializationFormats)); 
```

Also, what do you think about changing the behavior of this method a little bit so that a user can add multiple serialization formats via multiple calls?

```java
builder.otherSerializationFormats(formatA)
       .otherSerializationFormats(formatB);
```

Reply:
Sure, sounds good. Will make changes.